### PR TITLE
Api apdu

### PR DESCRIPTION
**Pull request by kingofpayne**

### CODE REVIEW

File: api/api.py
Comment (niooss-ledger) [inline api/api.py:188]: Why is the function locking the same condition twice?

File: api/api.py
Comment (niooss-ledger) [inline api/api.py:198]: * Maybe this code should also lock `self.condition` before setting `self.response`
* If `self.response` is not `None`, this means that no one is expecting the response or that a response was already received and will be overwritten. Showing a warning in Speculos console in such a case would help debug unexpected issues. What do you think?

**Review by greenknot — CHANGES_REQUESTED** [review]
Thanks a lot for the PR @kingofpayne!

File: api/api.py
Comment (kingofpayne) [reply api/api.py:188]: This is a mistake, I think first lock should be a separate `threading.Lock` since `wait` releases `condition` until response is available. Would it be ok for you ?

File: api/api.py
Comment (kingofpayne) [reply api/api.py]: Didn't know that! Much better indeed! Fixed.

File: tests/api/test_api.py
Comment (kingofpayne) [reply tests/api/test_api.py]: Fixed

File: api/api.py
Comment (niooss-ledger) [reply api/api.py:187]: What is the semantics of `self.condition`? There seems to be 2 possible meanings:
1. a lock on `self.response`, associated with signaling so that `exchange` can wait for `seph_apdu_callback` to receive response data.
2. a lock on the `/apdu` endpoint so that only one client can use it.

If the second meaning was expected, the current code is buggy, because the lock is released when `exchange` waits for the response data. A dedicated lock is then needed, for this second meaning.

Otherwise if the second meaning is not expected (is it possible to send another command before a response was received?), the comment `Lock for a command/response for one client` should be removed.

File: api/api.py
Comment (kingofpayne) [reply api/api.py]: Schema added.

File: api/swagger.yaml
Comment (kingofpayne) [reply api/swagger.yaml:15]: Agreed, changed!

File: api/api.py
Comment (kingofpayne) [reply api/api.py:187]: Both 1 and 2 are expected. `self.condition` should satisfy response waiting, and another `self.lock` should satisfy endpoint locking. I can commit a change if you do agree? (actually I had those two locks at the beginning and thought by mistake I could use only one).

File: api/api.py
Comment (kingofpayne) [reply api/api.py]: Fixed

File: api/api.py
Comment (kingofpayne) [reply api/api.py]: Fixed. Thank you niooss for helping fixing this part.

File: api/swagger.yaml
Comment (greenknot) [reply api/swagger.yaml:15]: The parameters are still passed in the query, aren't they?

**Review by greenknot — APPROVED** [review]
LGTM :+1: 

Please squash all your commits together and rebase on master before merging.

**Pull request merged**
